### DISCUSSION TITLE
用户未操作表单时，期望获取正确的表单 invalid 状态

### DISCUSSION THREAD

**Discussion by heaven2049** [alibaba/formily#1823] [Q&A 2.x] [answered]
目的：想实现提交按钮禁用状态与表单 invalid状态联动。
目前可使用 form.invalid 字段实现按钮禁用/非禁用。但在用户未操作时，invalid字段默认为false。

尝试在表单mount后手动触发validate，但此时用户未操作出现errors不符合预期。
手工 clearErrors 之后 form  invalid 值又变为true了（看了下源码，因为invalid字段由 errors计算）。

所以，在用户未操作表单时，有其他方式能获取到正确的invalid状态嘛？

**Comment by janryWang** [comment, marked as answer]
参考 https://github.com/alibaba/formily/discussions/1797 在onFormValuesChange的时候实时校验

**Reply by heaven2049** [reply]
感谢。确实是个不错的思路，相当于有个背景表单来做这个事情。
不过我们的场景是JSX方式的，理论上来说可以根据已有表单实例clone一份实例出来做校验。看了下源码，或许也可以改造当前的表单，增加一个校验方法如validateInbackend & backendErrors，

官方有计划支持这个能力嘛？

**Reply by janryWang** [reply]
纯jsx场景的话，不太好弄了，内置的话，性能肯定有问题，每次都实时全量校验

**Reply by heaven2049** [reply]
嗯，JSX场景clone一份Field构造表单出来，感觉性能也堪忧= =
